src/bots/mickeyBot.js

```diff
@@ -0,0 +1,213 @@
+            inline++;
+          } else {
+            gapIdx = colIdx + offset;
+            missCount++;
+          }
+
+          offset++;
+        }
+
+        if (inline === winLength - 1) {
+          if (conditions.checkGapWin(boardModel, colIdx, rowIdx, gapIdx)) {
+            winMove = gapIdx;
+          }
+
+          if (conditions.checkTailWin(boardModel, gameState, colIdx, rowIdx, offset, winLength)) {
+            winMove = winMove || colIdx + winLength - 1;
+          }
+
+          if (conditions.checkHeadWin(boardModel, gameState, colIdx, rowIdx)) {
+            winMove = winMove || colIdx - 1;
+          }
+        }
+      }
+    });
+  });
+
+  return winMove;
+}
+
+function checkForWinMove(boardModel, player, gameState, winLength) {
+  const colWin = checkColumnsWin(boardModel, player, gameState, winLength);
+  const rowWin = checkHorizontalOrDiagonalWin(boardModel, player, gameState, winLength, conditionChecks.row);
+  const upDiagonalWin = checkHorizontalOrDiagonalWin(boardModel, player, gameState, winLength, conditionChecks.upDiagonal);
+  const downDiagonalWin = checkHorizontalOrDiagonalWin(boardModel, player, gameState, winLength, conditionChecks.downDiagonal);
+
+  return colWin || rowWin || upDiagonalWin || downDiagonalWin;
+}
+
+function getMove(boardModel, gameState) { // TODO: Narrow down a strategy for choosing spaces when no wins or threats exist
+  const centerPosition = Math.floor(boardModel.length / 2);
+  let target = centerPosition;
+  let isFull = true;
+
+  while (isFull) {
+    if (target >= 0 && target < boardModel.length && boardModel[target].length < gameState.boardHeights[target]) {
+      isFull = false;
+      break;
+    }
+
+    target = randomMove(gameState);
+  }
+
+  return target;
+}
+
+function randomMove(gameState) {
+  return Math.floor(Math.random() * gameState.boardHeights.length);
+}
+
+// TODO: This is cycling through rows and columns an awful lot. Since the board is small, this doesn't matter much
+//   but this code does not scale well. Ideally, this would be refactored to cycle through columns and rows once and
+//   determine the winning, defensive, or strategic moves in ONE iteration
+function getNextMove(boardModel, player, gameState) {
+  const winLength = gameState.winCondition[0];
+
+  const win = checkForWinMove(boardModel, player, gameState, winLength); // try to win
+  const survive = checkForWinMove(boardModel, player === 0 ? 1 : 0, gameState, winLength); // block opponent from winning
+  const blockProgress = checkForWinMove(boardModel, player === 0 ? 1 : 0, gameState, winLength - 1); // block opponent from getting 3 in a row
+  const buildConsecutive = checkForWinMove(boardModel, player, gameState, winLength - 1); // try to get 3 in a row
+  const whatevs = getMove(boardModel, gameState); // place a token somewhere near the center
+
+  return win || survive || blockProgress || buildConsecutive || whatevs;
```

Comment:
Just FYI here, if you did those variables inline instead of precomputing them, then it would short circuit on the first true one and save time in the average case.